src/main/java/org/springframework/data/mapping/model/PersistentEntityParameterValueProvider.java

```diff
@@ -47,10 +48,14 @@ public PersistentEntityParameterValueProvider(PersistentEntity<?, P> entity, Pro
 	@SuppressWarnings("unchecked")
 	public <T> T getParameterValue(Parameter<T, P> parameter) {
 
-		var constructor = entity.getPersistenceConstructor();
+		var creator = entity.getEntityCreator();
 
-		if (constructor != null && constructor.isEnclosingClassParameter(parameter)) {
-			return (T) parent;
+		if (creator instanceof PreferredConstructor) {
```

Comment:
Could this be moved to a callback on `EntityCreator` with the default just returning `false`?

Reply:
Hum, enclosing class is only applicable in the context of constructors and non-static inner classes. We have a single call to that method so it doesn't seem to be a good fit for the general API.

Reply:
I thought about broadening the name of the method to something like `isParentParameter()` and have that notion of "enclosing class parameter" move into the `PersistenceConstructor` implementation. Methods actually have something similar to that synthetic parameter, called "receiver parameters" that don't appear on the call side of the invocation: https://www.logicbig.com/tutorials/core-java-tutorial/java-8-enhancements/explicit_receiver_parameters.html. Not saying we should support those, but it's not as conceptually odd as it seems.

Reply:
> thought about broadening the name of the method to something like isParentParameter()

That makes sense. Let me handle that is a separate commit.